### PR TITLE
feat: [AH-3049] Open source contributions documentation

### PR DESCRIPTION
**Pull request by VebjornG**
Since InDSL is generally available it should have documentation covering open source contributions. This PR aims to provide that.
[Ticket](https://cognitedata.atlassian.net/browse/AH-3049)

### CODE REVIEW

File: docs-source/source/contribute.rst [hunk 1]
```diff
@@ -29,8 +29,68 @@ guide focuses on the requirements to comply with it. Nevertheless, we accept any
 important and making our library more robust over time is of paramount importance. Please don't hesitate to submit a
 Github pull request for something as small as a typo.*
 
+Open source contributions
```
Comment (neringaalt) [inline docs-source/source/contribute.rst:32]: This is a repeating a bit what is stated above. 
I think we should remove the list as it is almost the same as above. 
Cognite Hub is described in the paragraph above as well. 

I like that you added Code of Conduct

File: docs-source/source/contribute.rst [hunk 2]
```diff
@@ -29,8 +29,68 @@ guide focuses on the requirements to comply with it. Nevertheless, we accept any
...
+    * Documentation improvements
+    * Sharing the project with others
+
+We encourage you to read this document to understand how to contribute to the project. If you have any questions, please
+reach out to us on `Cognite Hub <https://hub.cognite.com/>`_. We are happy to help you get started, and we welcome your
+efforts to improve InDSL as long as everyone involved is treated with respect. Cordiality is highly appreciated. 
+Please read our `Code of Conduct <https://indsl.docs.cognite.com/code_of_conduct.html>`_ before contributing.
+
+
+Code contributions
```
Comment (neringaalt) [inline docs-source/source/contribute.rst:49]: This paragraph can be merged with "Open source contributions" paragraph

File: docs-source/source/contribute.rst [hunk 3]
```diff
@@ -29,8 +29,68 @@ guide focuses on the requirements to comply with it. Nevertheless, we accept any
...
+------------------
+
+A good PR should be concise, clear, and easy to understand. In order to contribute, follow these steps:
+
+
+1. **Fork the repository**: Fork the `repository <https://github.com/cognitedata/indsl>`_ 
+to your own GitHub account.
+
+2. **Run the tests**: Confirm that the tests pass on your local machine. We use `pytest` for testing. 
+If they fail, investigate why they do. If you are unable to fix the issue, please reach out to us.
```
Comment (neringaalt) [inline docs-source/source/contribute.rst:59]:
```suggestion
If they fail and you are unable to fix the issue, please reach out to us.
```

File: docs-source/source/contribute.rst [hunk 4]
```diff
@@ -29,8 +31,57 @@ guide focuses on the requirements to comply with it. Nevertheless, we accept any
...
+
+
+1. **Fork the repository**: Fork the `repository <https://github.com/cognitedata/indsl>`_ 
+to your own GitHub account.
+
+2. **Run the tests**: Confirm that the tests pass on your local machine. We use `pytest` for testing. 
+If they fail and you are unable to fix the issue, please reach out to us.
+
+3. **Make your changes**: Make your changes to the code base. Make sure to follow the coding style and documentation guidelines.
+We use Flake8 to enforce the coding style, which in addition to pre-commit checks, will run automatically when you push your changes.
```
Comment (neringaalt) [inline docs-source/source/contribute.rst:53]:
```suggestion
 Pre-commit checks will run automatically when you push your changes.
```

File: docs-source/source/contribute.rst [hunk 5]
```diff
@@ -29,8 +31,57 @@ guide focuses on the requirements to comply with it. Nevertheless, we accept any
...
+You can also run pre-commit checks manually for all staged files by running ``poetry run pre-commit run --all-files``. 
+You can find the full list of formatting requirements specified in the
```
Comment (neringaalt) [inline docs-source/source/contribute.rst:55]:
```suggestion
```

File: docs-source/source/contribute.rst [hunk 6]
```diff
@@ -29,8 +31,57 @@ guide focuses on the requirements to comply with it. Nevertheless, we accept any
...
+to your own GitHub account.
+
+2. **Run the tests**: Confirm that the tests pass on your local machine. We use `pytest` for testing. 
+If they fail and you are unable to fix the issue, please reach out to us.
+
+3. **Make your changes**: Make your changes to the code base. Make sure to follow the coding style and documentation guidelines.
+We use Flake8 to enforce the coding style, which in addition to pre-commit checks, will run automatically when you push your changes.
+You can also run pre-commit checks manually for all staged files by running ``poetry run pre-commit run --all-files``. 
+You can find the full list of formatting requirements specified in the 
+`.pre-commit-config.yaml <https://github.com/cognitedata/indsl/blob/main/.pre-commit-config.yaml>`_ file in the root of the repository.
```
Comment (neringaalt) [inline docs-source/source/contribute.rst:56]:
```suggestion
```

File: docs-source/source/contribute.rst [hunk 7]
```diff
@@ -29,8 +31,57 @@ guide focuses on the requirements to comply with it. Nevertheless, we accept any
...
+We follow the Google Python Style Guide for docstrings.
+
+4. **Write tests**: If you are adding a new feature or fixing a bug, write tests using the `pytest` framework to cover the new code. 
+Make sure that they pass.
+
+5. **Make a pull request**: Once you are satisfied with your changes and all of the tests pass, make a pull request to 
+the main repository using the conventional commit message format.
```
Comment (neringaalt) [inline docs-source/source/contribute.rst:63]: add a link to indsl

**Comment by neringaalt** [comment]
@VebjornG could you fix all the places in the documentation where it says that repository should be cloned? We need to make sure that it should be done by forking repo for open source contributions, but Cognite people could clone it directly. Could you figure out how to phrase those sentences? Maybe with links for the open source contributions?

File: docs-source/source/contribute.rst (hunk 7 above)
Comment (VebjornG) [reply docs-source/source/contribute.rst:63]: That link is at the top of the "Contributing" section though

Comment (neringaalt) [reply docs-source/source/contribute.rst:63]: It can be confusing to know what we mean by "main repository"

Comment (VebjornG) [reply docs-source/source/contribute.rst]: Then i think it's better to change the wording here

**Comment by VebjornG** [comment]
> @VebjornG could you fix all the places in the documentation where it says that repository should be cloned? We need to make sure that it should be done by forking repo for open source contributions, but Cognite people could clone it directly. Could you figure out how to phrase those sentences? Maybe with links for the open source contributions?

Made a suggestion in the latest commit 👍🏻

File: docs-source/source/contribute.rst [hunk 8]
```diff
@@ -79,12 +128,14 @@ This project uses `Poetry <https://python-poetry.org/>`_ for dependency manageme
...
-   GitHub to your local environment.
+1. For open source contributions, fork the `InDSL <https://github.com/cognitedata/indsl>`_ main repository on
+   GitHub to your local environment. If you are a Cognite employee or this contribution is internal, 
+   you may clone the repository directly by switching out `yourusername` with 
+   `cognitedata`.
 
 .. prompt:: bash $
 
-    git clone git@github.com:cognitedata/indsl.git
+    git clone git@github.com:yourusername/indsl.git
```
Comment (neringaalt) [inline docs-source/source/contribute.rst:138]: do not change anything here

File: docs-source/source/contribute.rst [hunk 9]
```diff
@@ -79,12 +128,14 @@ This project uses `Poetry <https://python-poetry.org/>`_ for dependency manageme
    pip install poetry
 
 
-1. Clone the `InDSL <https://github.com/cognitedata/indsl>`_ main repository on
-   GitHub to your local environment.
+1. For open source contributions, fork the `InDSL <https://github.com/cognitedata/indsl>`_ main repository on
+   GitHub to your local environment. If you are a Cognite employee or this contribution is internal, 
+   you may clone the repository directly by switching out `yourusername` with 
+   `cognitedata`.
```
Comment (neringaalt) [inline docs-source/source/contribute.rst:134]:
```suggestion
   If this contribution is internal, 
   you may clone the repository directly.
```

File: docs-source/source/contribute.rst [hunk 10]
```diff
@@ -29,8 +31,55 @@ guide focuses on the requirements to comply with it. Nevertheless, we accept any
...
+
+3. **Make your changes**: Make your changes to the code base. Make sure to follow the coding style and documentation guidelines.
+Pre-commit checks will run automatically when you push your changes.
+You can also run pre-commit checks manually for all staged files by running ``poetry run pre-commit run --all-files``. 
+We follow the Google Python Style Guide for docstrings.
+
+4. **Write tests**: If you are adding a new feature or fixing a bug, write tests using the `pytest` framework to cover the new code. 
+Make sure that they pass.
+
+5. **Make a pull request**: Once you are satisfied with your changes and all of the tests pass, make a pull request
```
Comment (neringaalt) [inline docs-source/source/contribute.rst:60]: maybe `base repository` is the better wording

**Pull request merged**